src/librustc_mir/interpret/eval_context.rs

```diff
@@ -227,21 +201,23 @@ impl<'a, 'mir, 'tcx, M> InfiniteLoopDetector<'a, 'mir, 'tcx, M>
     pub fn observe_and_analyze(
         &mut self,
         machine: &M,
-        stack: &Vec<Frame<'mir, 'tcx>>,
         memory: &Memory<'a, 'mir, 'tcx, M>,
+        stack: &[Frame<'mir, 'tcx>],
     ) -> EvalResult<'tcx, ()> {
-        let snapshot = (machine, stack, memory);
 
-        let mut fx = FxHasher::default();
-        snapshot.hash(&mut fx);
-        let hash = fx.finish();
+        let mut hcx = memory.tcx.get_stable_hashing_context();
```

Comment:
Please use `self.tcx` unless there is a reason not to (and then document taht reason^^).

Reply:
Alright.